### PR TITLE
Odometry python support

### PR DESCRIPTION
**Pull request by DumDereDum**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: samples/python/odometry.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+
+import numpy as np
+import cv2 as cv
```
Comment (alalek) [inline samples/python/odometry.py:4]: Please add Python tests too for regular validation (`modules/3d/misc/python/test`)

Comment (DumDereDum) [reply samples/python/odometry.py:4]: done

File: modules/3d/misc/python/test/test_odometry.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+
+from tabnanny import check
+import numpy as np
+import cv2 as cv
+import unittest
+
```
Comment (alalek) [inline modules/3d/misc/python/test/test_odometry.py:7]: Where is it used?

File: modules/3d/misc/python/test/test_odometry.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+from tests_common import NewOpenCVTests
+
+class odometry_test(NewOpenCVTests):
+    def test_OdometryDefault(self):
+        depth = self.get_sample('cv/rgbd/depth.png', cv.IMREAD_ANYDEPTH).astype(np.float32)
+        #check rotation from 0 to 4 degree
+        for degree in range(0, 40, 5):
```
Comment (alalek) [inline modules/3d/misc/python/test/test_odometry.py:14]: Python tests should check Python bindings only.

Not the accuracy through passing of different input values.

---

Python test should be as fast as possible.

File: modules/3d/misc/python/test/test_odometry.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+
+from tabnanny import check
```
[the remark this answers is not in the document]
Comment (DumDereDum) [reply modules/3d/misc/python/test/test_odometry.py]: forgot to remove, now removed

File: modules/3d/misc/python/test/test_odometry.py (hunk 2 above)
Comment (DumDereDum) [reply modules/3d/misc/python/test/test_odometry.py]: removed

File: modules/3d/misc/python/test/test_odometry.py (hunk 3 above)
Comment (DumDereDum) [reply modules/3d/misc/python/test/test_odometry.py]: left only one test case for each algo

**Review by savuor — APPROVED** [review]

File: samples/python/odometry.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import sys
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--algo',
+        help="""DEPTH - works with depth,
+                RGB - works with images,
+                RGB_DEPTH - works with all,
+                defalt - runs all algos""",
```
Comment (alalek) [inline samples/python/odometry.py:16]: default

File: samples/python/odometry.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        '--destination_rgb_frame',
+        default="")
+
+    args = parser.parse_args()
+
+    depth1 = cv.imread(args.source_depth_frame, cv.IMREAD_ANYDEPTH).astype(np.float32)
+    depth2 = cv.imread(args.destination_depth_frame, cv.IMREAD_ANYDEPTH).astype(np.float32)
+
+    rgb1 = cv.imread(args.source_rgb_frame, cv.IMREAD_COLOR)
+    rgb2 = cv.imread(args.destination_rgb_frame, cv.IMREAD_COLOR)
```
Comment (alalek) [inline samples/python/odometry.py:41]: `cv.samples.findFile` should be used

File: samples/python/odometry.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        default="")
+    parser.add_argument(
+        '-dst_rgb',
+        '--destination_rgb_frame',
+        default="")
+
+    args = parser.parse_args()
+
+    depth1 = cv.imread(args.source_depth_frame, cv.IMREAD_ANYDEPTH).astype(np.float32)
+    depth2 = cv.imread(args.destination_depth_frame, cv.IMREAD_ANYDEPTH).astype(np.float32)
```
Comment (alalek) [inline samples/python/odometry.py:38]: These files are not used for "RGB" mode. There is no sense to request them from user.

Similar note is about rgb data.

File: samples/python/odometry.py (hunk 5 above)
Comment (DumDereDum) [reply samples/python/odometry.py:16]: done

File: samples/python/odometry.py (hunk 6 above)
Comment (DumDereDum) [reply samples/python/odometry.py]: done

File: samples/python/odometry.py (hunk 7 above)
Comment (DumDereDum) [reply samples/python/odometry.py]: done

**Comment by DumDereDum** [comment]
@alalek, sorry for the bother 
changes done. may you merge this PR?

**Pull request merged**